### ISSUE TITLE
No graph if time less than 5s

### ISSUE THREAD

**Issue by albfan** [DaveDavenport/gpx-viewer#27]
https://github.com/DaveDavenport/gpx-viewer/blob/master/src/gpx-viewer.c#L612

Some gpx contains incorrect time.

With an option on custom dialog, it can create a fake timestamps and graphs are available.

Some graphs like elevation or distance could exists without this condition.

a speed alghoritm depending on elevation percentage could create accurate timestamps for walk, run, bycicle, moto....

**Comment by albfan** [comment]
Some people uoload tracks to webs like Wikiloc, where poibts are correcto, but time is fake.

Maybe check time and distance is a better check.

In that case, time is not accurate, but route, elevation are still valid info.

**Comment by albfan** [comment]
here two gpx:

https://es.wikiloc.com/rutas-ciclismo/campomanes-la-cubilla-campomanes-28067730

original: (indented wth xmllint):
campomanes-la-cubilla-campomanes_format.gpx 

and what I rework (fake time each 10s and removing go back from track)
campomanes-la-cubilla.gpx

[gpx.zip](https://github.com/DaveDavenport/gpx-viewer/files/9242819/gpx.zip)

the speed alghoritm should slow when hiking, and go fast descending. (but that's different issue)

**Comment by DaveDavenport** [comment]
That gpx file is very broken, it has time jumping 3? hours  to the past suddenly. this screws up the internal cleaning algorithm.

Not sure if this should be supported?

**Comment by albfan** [comment]
Of course, gpx is the problem, but looks typical (is this comes from a gpx service like wikiloc). Looks people take care of position and less about time.

Including some graphs are unrelated with time (elevation, total distance) after that probably you want to fix the time.

But again that's my intention to turn gpx viewer into a cleaner, editor, which probably is above the gpx-viewer scope.

In branch fake-time:

https://github.com/DaveDavenport/gpx-viewer/tree/fake-time

there's a result to ignore time from gpx and create a new "fake" one, that can be saved (this is the code base to create the cleanup campomanes-la-cubilla.gpx (shortening the track and fixing the time)

**Comment by DaveDavenport** [comment]
I don't agree that time is unrelated in a gpx file.  its one of two axis we show, so when time is broken it not showing a graph (because one axis is broken) seems valid to me. This also goes for the elevation and distance graph.

Having said that, if we can correctly identify a broken gpx file upon loading and popup a 'do you want to fix this' dialog, I think this would be an acceptable solution.

**Comment by albfan** [comment]
Sounds good to me, let me work on an isolate PR.

That opens the viewer for algorithms to recreate time.

gpsbabel has faketime (specify first time and add an increment) http://www.gpsbabel.org/htmldoc-development/filter_track.html

but more accurate should be a constant speed (walk, bycicle, car) and some correction factor depending on hill grade (uphill or downhill)

**Comment by DaveDavenport** [comment]
I think I now finally understand your first post.
